### DISCUSSION TITLE
How to show message in custom managed bootstrap application

### DISCUSSION THREAD

**Discussion by radz1** [wixtoolset/issues#6325] [Questions] [answered]
Hi,

I am changing my standard bootstrapper application with customised managed bootstrapper. 

I have below conditional message based on the registry search. 

How do I implement with customized managed bootstrapper. any example would be appreciated. 

```
<Variable Name="NetFrameworkKey" Value="SOFTWARE\Microsoft\NET Framework Setup\NDP\v4\Full" Type="string"/>
    <util:RegistrySearch Root="HKLM" Key="[NetFrameworkKey]" Result="value" Variable="NetFramework_x64_Installed_48x" Win64="yes" Format="raw" Value="Version"/>
    <bal:Condition Message="This application requires .NET Framework 4.8 or later. Please install .NET Framework and restart the machine. After restart run this installer again.   Download &lt;a href=&quot;https://dotnet.microsoft.com/download/dotnet-framework/thank-you/net48-web-installer &quot;&gt;.NET Framework 4.8 &lt;/a&gt;.">
      <![CDATA[Installed OR 
          (NetFramework_x64_Installed_48x >= "4.8.0")]]>
    </bal:Condition>
```
Thanks

**Comment by robmen** [comment, marked as answer]
The WiX Toolset is Open Source which means you can see the source code how everything was implemented. The code for evaluating bootstrapper conditions is here: https://github.com/wixtoolset/wix3/blob/develop/src/libs/balutil/balcondition.cpp

**Reply by radz1** [reply]
Hi rob, 

I am very new to wix and did not find good documentation for customizing the bootstrapper.  I understand we can see the code implementation but wanted to know whether the construct can still would be used in managed bootstrapper or we need to implement it

As in case of standard bootstrapper application, we had <util:RegistrySearch> and <bal:condition> construct in wix file and the message is shown in the screen if it fails the condition. 

My question is does this construct <util:RegistrySearch> and <bal:condition> will work with  "BootstrapperApplicationRef Id="ManagedBootstrapperApplicationHost"" ? 

If yes, how to show failed message  on custom screen ? like which event needs to be handle and property have the error message. 

Thanks,

**Reply by robmen** [reply]
Searches are executed automatically by Burn. The code I linked to evaluates the condition and provides a message. Displaying the message is dependent on your UI stack but should be pretty trivial.
